dask/dataframe/core.py

```diff
@@ -5127,6 +5128,10 @@ def set_index(
         inplace: bool, optional
             Modifying the DataFrame in place is not supported by Dask.
             Defaults to False.
+        sort: bool, optional
+            Determine new divisions and repartition according to new index, otherwise
+            simply set the index on the individual existing partitions.
+            Defaults to True.
```

Comment:
As it turns out my description in the issue was wrong and we are in fact _sorting_ the entire dataframe, not just repartitioning according to the new divisions (https://github.com/dask/dask/blob/cb7d78051f53e6fd8ab624f44dd6960bd2433d46/dask/dataframe/shuffle.py#L347). Sorry! Therefore I think we can simplify this description:

```suggestion
            Whether to sort the dataframe by the new index.
            Defaults to True.
```